migration/v0.22.x-v0.23.x/user-alertmanager-reconfig.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+set -euo pipefail
+
+: "${CK8S_CONFIG_PATH:?Missing CK8S_CONFIG_PATH}"
+
+here="$(dirname "$(readlink -f "${BASH_SOURCE[0]}")")"
+
+alertmanager_config_bk="${CK8S_CONFIG_PATH}/user-alertmanager.yaml.bk"
+alertmanager_config_new="${CK8S_CONFIG_PATH}/user-alertmanager.yaml"
+common_config="${CK8S_CONFIG_PATH}/common-config.yaml"
+wc_config="${CK8S_CONFIG_PATH}/wc-config.yaml"
+sc_config="${CK8S_CONFIG_PATH}/sc-config.yaml"
+
+cleanup(){
+  echo -n "Do you want to remove the user-alertmanager config backups? [y/N]: "
+  read -r reply
+  if [[ "${reply}" == "y" ]]; then
+      rm "$@"
+  fi
+}
+
+if [[ ! -f "${common_config}" ]]; then
+    echo "Override common-config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${wc_config}" ]]; then
+    echo "Override wc-config does not exist, aborting."
+    exit 1
+elif [[ ! -f "${sc_config}" ]]; then
+    echo "Override sc-config does not exist, aborting."
+    exit 1
+fi
+
+wc=$(yq m -x -a overwrite -j "${common_config}" "${wc_config}" "${sc_config}" | yq r - 'user.alertmanager')
```

Comment:
Overall it seems to be working fine. But I think you can skip merging in the `sc-config.yaml` as it isn't needed here.

Reply:
done